crates/astria-conductor/src/driver.rs

```diff
@@ -243,3 +266,466 @@ impl Driver {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use test_utils::{
+        get_test_config,
+        MockCelestia,
+        MockExecution,
+        MockSequencer,
+    };
+
+    use super::*;
+    use crate::config::CommitLevel;
+
+    #[tokio::test]
+    async fn new_driver_execution_commit_level_set_to_soft_only() {
+        let mut config = get_test_config();
+        config.execution_commit_level = CommitLevel::SoftOnly;
+
+        let execution_server = MockExecution::spawn().await;
+        config.execution_rpc_url = execution_server.local_addr();
+
+        let sequencer = MockSequencer::spawn().await;
+        config.sequencer_url = sequencer.local_addr();
+
+        let (driver, ..) = Driver::new(config).await.unwrap();
+
+        assert!(driver.reader_tx.is_none());
+        assert!(driver.sequencer_client.is_some());
+    }
+
+    #[tokio::test]
+    async fn new_driver_execution_commit_level_set_to_firm_only() {
+        let mut config = get_test_config();
+        config.execution_commit_level = CommitLevel::FirmOnly;
+
+        let execution_server = MockExecution::spawn().await;
+        config.execution_rpc_url = execution_server.local_addr();
+
+        let mut celestia = MockCelestia::spawn().await;
+        config.celestia_node_url = celestia.local_addr();
+
+        let (driver, ..) = Driver::new(config).await.unwrap();
+
+        assert!(driver.reader_tx.is_some());
+        assert!(driver.sequencer_client.is_none());
+    }
+
+    #[tokio::test]
+    async fn new_driver_execution_commit_level_set_to_soft_and_firm() {
+        let mut config = get_test_config();
+        config.execution_commit_level = CommitLevel::SoftAndFirm;
+
+        let execution_server = MockExecution::spawn().await;
+        config.execution_rpc_url = execution_server.local_addr();
+
+        let sequencer = MockSequencer::spawn().await;
+        config.sequencer_url = sequencer.local_addr();
+
+        let mut celestia = MockCelestia::spawn().await;
+        config.celestia_node_url = celestia.local_addr();
+
+        let (driver, ..) = Driver::new(config).await.unwrap();
+
+        assert!(driver.reader_tx.is_some());
+        assert!(driver.sequencer_client.is_some());
+    }
+}
+
+#[cfg(test)]
+pub(crate) mod test_utils {
```

Comment:
this mod only appears to be used in the tests above, any reason to not combine them? or move this mod to its own file?

Reply:
I'll move this to it's own file. 
There was a lot of code to get the mocks working and I wanted to reorganize things so that the test mod was cluttered